plugins/algoltek-usb/fu-algoltek-usb-device.c

```diff
@@ -91,11 +93,30 @@ fu_algoltek_usb_device_rdv(FuAlgoltekUsbDevice *self, GError **error)
 						  error))
 		return NULL;
 
-	/* Remove len and cmd bytes */
-	for (guint32 i = 2; i < st->len; i++) {
-		if (st->data[i] < 128)
-			fu_byte_array_append_uint8(version_data, st->data[i]);
+	if (!fu_memread_uint16_safe(st->data, st->len, 2, &version_prefix, G_BIG_ENDIAN, error))
+		return NULL;
+
+	if (version_prefix == 0x4147) {
+		/* remove len, cmd bytes and "AG" prefixes */
+		for (guint32 i = 4; i < st->len; i++) {
+			if (st->data[i] == '_') {
+				underscore_count++;
+				if (underscore_count == 1)
+					continue;
+			}
+			if (underscore_count > 2)
+				break;
+			if (underscore_count > 0)
```

Comment:
but you can remove this one, right? it's never going to be ==0

Reply:
There are some numbers between the 'AG' prefixes and the first '_', so the underscore_count maybe 0.

And the count of those numbers isn't a constant, so I need to check each char.

Reply:
okay, no worries, thanks,.